Source/Core/Core/PowerPC/MMU.h

```diff
@@ -15,14 +15,55 @@ namespace PowerPC
 {
 // Routines for debugger UI, cheats, etc. to access emulated memory from the
 // perspective of the CPU.  Not for use by core emulation routines.
-// Use "Host_" prefix.
+// Use "Host" prefix.
+
+enum class RequestedAddressSpace
+{
+  Effective,  // whatever the current MMU state is
+  Physical,   // as if the MMU was turned off
+  Virtual,    // specifically want MMU turned on, returns std::nullopt if off
+};
+
 u8 HostRead_U8(u32 address);
 u16 HostRead_U16(u32 address);
 u32 HostRead_U32(u32 address);
 u64 HostRead_U64(u32 address);
 float HostRead_F32(u32 address);
 double HostRead_F64(u32 address);
 u32 HostRead_Instruction(u32 address);
+std::string HostGetString(u32 address, size_t size = 0);
+
+template <typename T>
+struct TryReadResult
+{
+  bool success;
+  bool translated;
+  T value;
+
+  TryReadResult() : success(false) {}
+  TryReadResult(bool translated_, T value_) : success(true), translated(translated_), value(value_)
+  {
+  }
+  operator bool() const { return success; }
+};
+
+TryReadResult<u8> HostTryReadU8(u32 address,
```

Comment:
AFAICT all these functions that use `TryReadResult` are only used in the cpp, could they be moved to file local static (as well as `TryReadResult` itself) or are you planning on using it elsewhere later?

Reply:
`TryReadResult` is going to be used from elsewhere in the future, yeah.